### DISCUSSION TITLE
Single session per user

### DISCUSSION THREAD

**Discussion by RobinBaeckman** [ory/hydra#2661] [Q&A] [answered]
I need to setup so if a user session already exists then give error letting user know they are already signed in. How can I check if session for a subject already exists?

**Comment by aeneasr** [comment, marked as answer]
https://www.ory.sh/kratos/docs/self-service/hooks/#revoke_active_sessions

**Reply by RobinBaeckman** [reply]
the criteria is that the attempt of the second login should be blocked, and this is for hydra. I'm looking at "Lists All Consent Sessions of a Subject" from the api, seems to be the only way to see if a user already has a session by using the subject id?

**Reply by aeneasr** [reply]
Sorry, my bad, I definitely was thinking this was in Ory Kratos. You would have to do this in your consent app. There you could, for example, invalidate all prior consents.

Please keep in mind that OAuth2 is not login: https://www.ory.sh/hydra/docs/concepts/before-oauth2#access-and-refresh-tokens-are-not-sessions
